pkg/sdk/stages_dto_gen.go

```diff
@@ -0,0 +1,263 @@
+	_ optionsProvider[CreateOnS3CompatibleStageOptions]    = new(CreateOnS3CompatibleStageRequest)
+	_ optionsProvider[AlterStageOptions]                   = new(AlterStageRequest)
+	_ optionsProvider[AlterInternalStageStageOptions]      = new(AlterInternalStageStageRequest)
+	_ optionsProvider[AlterExternalS3StageStageOptions]    = new(AlterExternalS3StageStageRequest)
+	_ optionsProvider[AlterExternalGCSStageStageOptions]   = new(AlterExternalGCSStageStageRequest)
+	_ optionsProvider[AlterExternalAzureStageStageOptions] = new(AlterExternalAzureStageStageRequest)
+	_ optionsProvider[AlterDirectoryTableStageOptions]     = new(AlterDirectoryTableStageRequest)
+	_ optionsProvider[DropStageOptions]                    = new(DropStageRequest)
+	_ optionsProvider[DescribeStageOptions]                = new(DescribeStageRequest)
+	_ optionsProvider[ShowStageOptions]                    = new(ShowStageRequest)
+)
+
+type CreateInternalStageRequest struct {
+	OrReplace             *bool
+	Temporary             *bool
+	IfNotExists           *bool
+	name                  SchemaObjectIdentifier // required
+	Encryption            *InternalStageEncryptionRequest
+	DirectoryTableOptions *InternalDirectoryTableOptionsRequest
+	FileFormat            *StageFileFormatRequest
+	CopyOptions           *StageCopyOptionsRequest
+	Comment               *string
+	Tag                   []TagAssociation
+}
+
+type InternalStageEncryptionRequest struct {
+	Type *InternalStageEncryptionOption // required
+}
+
+type InternalDirectoryTableOptionsRequest struct {
+	Enable          *bool
+	RefreshOnCreate *bool
+}
+
+type StageFileFormatRequest struct {
+	FormatName *string
+	Type       *FileFormatType
+	Options    *FileFormatTypeOptions
+}
+
+type StageCopyOptionsRequest struct {
+	OnError           *StageCopyOnErrorOptionsRequest
+	SizeLimit         *int
+	Purge             *bool
+	ReturnFailedOnly  *bool
+	MatchByColumnName *StageCopyColumnMapOption
+	EnforceLength     *bool
+	Truncatecolumns   *bool
+	Force             *bool
+}
+
+type StageCopyOnErrorOptionsRequest struct {
+	Continue       *bool
+	SkipFile       *bool
+	AbortStatement *bool
+}
+
+type CreateOnS3StageRequest struct {
+	OrReplace             *bool
+	Temporary             *bool
+	IfNotExists           *bool
+	name                  SchemaObjectIdentifier // required
+	ExternalStageParams   *ExternalS3StageParamsRequest
+	DirectoryTableOptions *ExternalS3DirectoryTableOptionsRequest
+	FileFormat            *StageFileFormatRequest
+	CopyOptions           *StageCopyOptionsRequest
+	Comment               *string
+	Tag                   []TagAssociation
+}
+
+type ExternalS3StageParamsRequest struct {
+	Url                string // required
+	StorageIntegration *AccountObjectIdentifier
+	Credentials        *ExternalStageS3CredentialsRequest
+	Encryption         *ExternalStageS3EncryptionRequest
+}
+
+type ExternalStageS3CredentialsRequest struct {
+	AwsKeyId     *string
```

Comment:
minor point, but Aws is an acronymn that stands for Amazon Web Services. I don't think it should ever be written as Aws. It should be AWSKeyId. If this is an intentional design decision for readability, then we should agree on it.

Reply:
Sure, I've changed it to uppercase. I've talked with @sfc-gh-asawicki and maybe we could use uppercased version, because it's used in other places like CSV, JSON and XML in the file format files.